### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by miki-yuasa**
Added a new environment. No changes to other files.

### CODE REVIEW

File: gym_multigrid/envs/maze.py [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+
+from gym_multigrid.core.agent import Agent, AgentT
+from gym_multigrid.core.grid import Grid
+from gym_multigrid.core.object import Floor, Flag, Obstacle, WorldObjT
+from gym_multigrid.core.world import World
+from gym_multigrid.multigrid import MultiGridEnv
+from gym_multigrid.typing import Position
+from gym_multigrid.utils.map import distance_area_point
+
+MazeColors: dict[str, NDArray] = {
```
Comment (apurl1) [inline gym_multigrid/envs/maze.py:16]: can we put this in core/constants.py?

File: gym_multigrid/envs/maze.py [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+
+class MazeActions(enum.IntEnum):
+    stay = 0
+    left = 1
+    down = 2
+    right = 3
+    up = 4
+
+
+MazeWorld = World(
```
Comment (apurl1) [inline gym_multigrid/envs/maze.py:39]: can we put this in core/world.py?

**Review by apurl1 — CHANGES_REQUESTED** [review]
good overall. left suggestions in comments to move some definitions to more appropriate code files.

File: gym_multigrid/envs/maze.py (hunk 1 above)
Comment (miki-yuasa) [reply gym_multigrid/envs/maze.py]: Moved to core/constants.py

File: gym_multigrid/envs/maze.py (hunk 2 above)
Comment (miki-yuasa) [reply gym_multigrid/envs/maze.py]: Moved

**Review by apurl1 — APPROVED** [review]

**Pull request merged**